routes/routes_impl.py

```diff
@@ -615,6 +622,45 @@ def bathymetry_impl(projection: str, zoom: int, x: int, y: int):
         img = plotting.tile.bathymetry(projection, x, y, zoom, {})
         return _cache_and_send_img(img, f)
 
+def mbt_impl(projection: str, tiletype: str, zoom: int, x: int, y: int):
+  """
+       Serves mbt files
+  """
+  cache_dir = current_app.config['CACHE_DIR']
+  shape_file_dir = current_app.config['SHAPE_FILE_DIR']
+  directory = str(os.path.join(cache_dir, request.path[1:]))
+  # This is a funky way of splitting the string to only include characters up to the 10th instance of '/'
+  basedir = "/".join(directory.split("/", 10)[:10])
+
+  # Send blank tile if conditions aren't met
+  if (zoom < 7) or (projection != "EPSG:3857"):
+    return send_file(shape_file_dir + "/blank.mbt")
+
+# Send file if cached or select data in SQLite file
+  if os.path.isfile(directory):
+    return send_file(directory)
+  else:
+    y = (2**zoom-1) - y
+    connection = sqlite3.connect(shape_file_dir + "/{}.mbtiles".format(tiletype))
+    selector = connection.cursor()
+    sqlite = "SELECT tile_data FROM tiles WHERE zoom_level = {} AND tile_column = {} AND tile_row = {}".format(zoom, x, y)
+    selector.execute(sqlite)
+    tile = selector.fetchone()
+    if tile == None:
+        return send_file(shape_file_dir + "/blank.mbt")
+
+    # Write tile to cache and send file
+    if os.path.isdir(basedir):
```

Comment:
Similar comment as earlier, we can simplify this like so:

```python
if not os.path.isdir(basedir):
  os.makedirs(basedir)
```

Reply:
Changed this and the previous mention of this.